src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/device-list/device-list.component.ts

```diff
@@ -0,0 +1,73 @@
+import { DatePipe } from '@angular/common';
+import { Component, Input, OnInit, TemplateRef, ViewChild } from '@angular/core';
+import { I18n } from '@ngx-translate/i18n-polyfill';
+import { HostService } from '../../../shared/api/host.service';
+import { OsdService } from '../../../shared/api/osd.service';
+import { CdTableColumn } from '../../../shared/models/cd-table-column';
+import { CdDevice } from '../../../shared/models/devices';
+
+@Component({
+  selector: 'cd-device-list',
+  templateUrl: './device-list.component.html',
+  styleUrls: ['./device-list.component.scss']
+})
+export class DeviceListComponent implements OnInit {
+  @Input()
+  hostname = '';
+  @Input()
+  osdId = null;
+
+  @ViewChild('deviceLocation', { static: true })
+  locationTemplate: TemplateRef<any>;
+  @ViewChild('lifeExpectancy', { static: true })
+  lifeExpectancyTemplate: TemplateRef<any>;
+  @ViewChild('lifeExpectancyTimestamp', { static: true })
+  lifeExpectancyTimestampTemplate: TemplateRef<any>;
+  @ViewChild('state', { static: true })
+  stateTemplate: TemplateRef<any>;
+
+  devices: CdDevice[] = null;
+  columns: CdTableColumn[] = [];
+  translationMapping = {
+    '=1': '# week',
+    other: '# weeks'
+  };
+
+  constructor(
+    private hostService: HostService,
+    private i18n: I18n,
+    private datePipe: DatePipe,
+    private osdService: OsdService
+  ) {}
+
+  ngOnInit() {
+    const updateDevices = (devices) => (this.devices = devices);
```

Comment:
Please name the variable like below to see that it holds a function. This will help to understand the code.
```suggestion
    const updateDevicesFn = (devices) => (this.devices = devices);
```
or
```suggestion
    const updateDevicesFunc = (devices) => (this.devices = devices);
```